scripts/extensions/auto-tagging-widget/src/ImageTaggingComponent/ImageTaggingComponent.tsx

```diff
@@ -0,0 +1,355 @@
+  style?: React.CSSProperties;
+  article: IArticle;
+}
+
+interface IState {
+  showImages: boolean;
+  isLoading: boolean;
+  fetchError: boolean;
+  selectedImage: IImage | null;
+  images: Array<IImage>;
+  prevInput: Array<ITagInput>;
+}
+
+const gridContainerStyle: React.CSSProperties = {
+  display: "grid",
+  gridTemplateColumns: "3fr 1fr",
+  gridColumnGap: "4px",
+  maxHeight: "320px",
+  overflow: "hidden",
+};
+
+const gridItemLeftStyle: React.CSSProperties = {
+  width: "100%",
+  height: "100%",
+  maxHeight: "inherit",
+};
+
+const gridItemRightStyle: React.CSSProperties = {
+  width: "100%",
+  height: "100%",
+  maxHeight: "inherit",
+};
+
+const imageContentStyle: React.CSSProperties = {
+  width: "100%",
+  maxHeight: "inherit",
+  overflowY: "auto",
+  display: "grid",
+  gridRowGap: "4px",
+};
+
+const imageStyle: React.CSSProperties = {
+  width: "100%",
+  height: "auto",
+};
+
+const imageWrapperStyle: React.CSSProperties = {
+  cursor: "pointer",
+  maxHeight: "100%",
+};
+
+const selectedCardStyle: React.CSSProperties = {
+  display: "flex",
+  flexDirection: "column",
+  height: "100%",
+  overflow: "hidden",
+  boxShadow: "1px 3px 6px 0 rgba(0,0,0,0.2)",
+  width: "100%",
+  borderRadius: "4px",
+};
+
+const cardStyle: React.CSSProperties = {
+  boxShadow: "1px 3px 6px 0 rgba(0,0,0,0.2)",
+  width: "100%",
+  borderRadius: "2px",
+};
+
+export function getImageTaggingComponent(
+  superdesk: ISuperdesk,
+): React.ComponentType<IProps> {
+  const { httpRequestJsonLocal } = superdesk;
+
+  return class ImageTagging extends React.PureComponent<IProps, IState> {
+    private _mounted: boolean;
+    private abortController: AbortController;
+    private fetchTimeout: ReturnType<typeof setTimeout> | undefined;
+    constructor(props: IProps) {
+      super(props);
+
+      this.state = {
+        showImages: true,
+        isLoading: false,
+        fetchError: false,
+        selectedImage: null,
+        images: [],
+        prevInput: [],
+      };
+      this._mounted = false;
+      this.abortController = new AbortController();
+      this.fetchTimeout = undefined;
+      this.runFetchImages = this.runFetchImages.bind(this);
+      this.formatTags = this.formatTags.bind(this);
+      this.handleClickImage = this.handleClickImage.bind(this);
+      this.handleDragStart = this.handleDragStart.bind(this);
+    }
+
+    componentDidMount() {
+      this._mounted = true;
+      if (!this.state.isLoading) {
+        this.runFetchImages();
+      }
+    }
+
+    componentWillUnmount() {
+      this._mounted = false;
+      this.abortController.abort();
+      clearTimeout(this.fetchTimeout);
+    }
+
+    componentDidUpdate() {
+      clearTimeout(this.fetchTimeout);
+      if (
+        this._mounted &&
+        !this.state.isLoading &&
+        !this.isEqualTags(
+          this.formatTags(this.props.data),
+          this.state.prevInput,
+        )
+      ) {
+        this.fetchTimeout = setTimeout(() => {
```

Comment:
The purpose of the timeout is to prevent immediate fetch upon tag changes. E.g. when manually removing 2 tags really quickly, we do not want 2 parallel requests. This is one way of preventing this without needing to cancel the first request.

Reply:
Can you use `debounce` or `throttle` from lodash instead? It accomplishes the same, but uses a more standard approach.